client-common/src/tendermint/mock.rs

```diff
@@ -0,0 +1,301 @@
+#![allow(missing_docs)]
+use crate::tendermint::types::*;
+use serde_json;
+use std::str::FromStr;
+use tendermint::{
+    account, chain, channel, net, node, validator, Moniker, PrivateKey, PublicKey, Version,
+};
+
+const DEFAULT_VALIDATOR_KEY: &str = "{
+  \"type\": \"tendermint/PrivKeyEd25519\",
+  \"value\": \"gJWgIetdLxRc/C2t/XjV65NCqZLTqHS9pU69kBzRmyOKCHDqT/6bKPmKajdBp+KCbYtu9ttTX7+MXrEQOw8Kqg==\"
+}";
+const BLOCK_VERSION: u64 = 10;
+const APP_VERSION: u64 = 0;
+const DEFAULT_GENESIS_JSON: &str = r#"{
```

Comment:
Not sure if it'll make this a bit nicer / more readable -- these long strings for testing could be put into separate files (e.g. `mock/genesis.json`) and added here using a macro (e.g. `include_str!("mock/genesis.json");` )

Reply:
I was trying to use the `Default` trait here, which would save some code, but arbitrary `Default` implementation doesn't always make sense, and the PR of this idea gets rejected upstream.

Also, I marked ignore to one of the tests of manual synchronizer, because of the problem of test data. Now with fully verified synchronizer, we need to mock full verifiable block data, which is not done so far. Maybe we should create a dedicated crate for this, which only included in dev-dependencies.